plugins/outputs/sematext/processors/rename.go

```diff
@@ -0,0 +1,114 @@
+		"nginx.reading":  "nginx.requests.connections.reading",
+		"nginx.writing":  "nginx.requests.connections.writing",
+		"nginx.waiting":  "nginx.requests.connections.waiting",
+		"nginx.requests": "nginx.request.count",
+		// mongodb
+		"mongo.flushes":                   "mongo.flushes",
+		"mongo.flushes_total_time_ns":     "mongo.flushes.time",
+		"mongo.document_inserted":         "mongo.documents.inserted",
+		"mongo.document_updated":          "mongo.documents.updated",
+		"mongo.document_deleted":          "mongo.documents.deleted",
+		"mongo.document_returned":         "mongo.documents.returned",
+		"mongo.resident_megabytes":        "mongo.memory.resident",
+		"mongo.vsize_megabytes":           "mongo.memory.virtual",
+		"mongo.mapped_megabytes":          "mongo.memory.mapped",
+		"mongo.inserts":                   "mongo.ops.insert",
+		"mongo.queries":                   "mongo.ops.query",
+		"mongo.updates":                   "mongo.ops.update",
+		"mongo.getmores":                  "mongo.ops.getmore",
+		"mongo.commands":                  "mongo.ops.command",
+		"mongo.repl_inserts":              "mongo.replica.ops.insert",
+		"mongo.repl_queries":              "mongo.replica.ops.query",
+		"mongo.repl_updates":              "mongo.replica.ops.update",
+		"mongo.repl_deletes":              "mongo.replica.ops.delete",
+		"mongo.repl_getmores":             "mongo.replica.ops.getmore",
+		"mongo.repl_commands":             "mongo.replica.ops.command",
+		"mongo.count_command_failed":      "mongo.commands.failed",
+		"mongo.count_command_total":       "mongo.commands.total",
+		"mongo.data_size":                 "mongo.database.data.size",
+		"mongo.storage_size":              "mongo.database.storage.size",
+		"mongo.index_size":                "mongo.database.index.size",
+		"mongo.collections":               "mongo.database.collections",
+		"mongo.objects":                   "mongo.database.objects",
+		"mongo.connections_current":       "mongo.network.connections",
+		"mongo.connections_total_created": "mongo.network.connections.total",
+		"net_in_bytes":                    "mongo.network.transfer.rx.rate",
+		"net_out_bytes":                   "mongo.network.transfer.tx.rate",
+	}
+)
+
+// Rename processor renames the measurement (metric) names
+// to match the existing metric names sent by Node.js agents
+type Rename struct{}
+
+// NewRename builds a new rename processor.
+func NewRename() BatchProcessor { return &Rename{} }
+
+// Process performs a lookup in the local maps of metric/field names
+// and replaces the metric name with the new name.
+func (r *Rename) Process(points []telegraf.Metric) ([]telegraf.Metric, error) {
+	for _, point := range points {
+		replace, ok := measurementReplaces[point.Name()]
+		if ok {
+			point.SetName(replace)
+		}
+		for _, field := range point.FieldList() {
+			key := point.Name() + "." + field.Key
+			replace, ok := fieldReplaces[key]
+			if !ok {
```

Comment:
So, if don't have a mapping, we change the namespace, but keep the key. We'll build future charts/reports based on such namespace.key combo.

Reply:
Committed an adjustment.